docs/language/contracts.md

```diff
@@ -261,12 +284,107 @@ let code = "70756220636f6e...".decodeHex()
 // `code` has type `[UInt8]`
 
 let signer: Account = ...
-signer.setCode(
-    code,
+signer.contracts.add(
+    name: "Test"
+    code: code,
     message: "I'm a new contract in an existing account"
 )
 ```
 
+#### Updating a Deployed Contract
+
+> 🚧 Status: Updating contracts is **experimental**.
+
+A deployed contract can be updated using the `update__experimental` function:
+
+  ```cadence
+  fun update__experimental(name: String, code: [UInt8]): DeployedContract
+  ```
+
+  Updates the code for the contract/contract interface in the account.
+
+  The `code` parameter is the UTF-8 encoded representation of the source code.
+  The code must contain exactly one contract or contract interface,
+  which must have the same name as the `name` parameter.
+
+  Does **not** run the initializer of the contract/contract interface again.
+  The contract instance in the world state stays as is.
+
+  Fails if no contract/contract interface with the given name exists in the account,
+  if the given code does not declare exactly one contract or contract interface,
+  or if the given name does not match the name of the contract/contract interface declaration in the code.
+
+  Returns the deployed contract for the updated contract.
+
+For example, assuming that a contract named `Test` is already deployed to the account
+and it should be updated with the following contract code:
+
+```cadence,file=test-contract.cdc
+pub contract Test {
+    pub let message: String
+
+    init(message: String) {
+        self.message = message
+    }
+}
+```
+
+The contract can be updated as follows:
+
+```cadence,file=account-contracts-update.cdc
+// Decode the hex-encoded source code into a byte array
+// using the built-in function `decodeHex`.
+//
+// (The ellipsis ... indicates the remainder of the string)
+//
+let code = "70756220636f6e...".decodeHex()
+
+// `code` has type `[UInt8]`
+
+let signer: Account = ...
+signer.contracts.update__experimental(name: "Test", code: code)
+```
+
+#### Getting a Deployed Contract
+
+A deployed contract can be get from an account using the `get` function:
+
+  ```cadence
+  fun get(name: String): DeployedContract?
```

Comment:
I see some examples here referencing `DeployedContract` but I don't see the definition for `DeployedContract`. Should it be `DeployedCode` instead?

Reply:
Ah, I had changed to type annotations from `DeployedCode` to `DeployedContract`, but forgot to rename the struct definition below, good catch! 👍

Reply:
Fixed in 130ed31